src/MCPClient/lib/clientScripts/verify_checksum.py

```diff
@@ -0,0 +1,276 @@
+# -*- coding: utf-8 -*-
+
+"""Verify Checksum Job
+
+Archivematica wraps the hashsum utility to verify checksums provided to the
+system as part of a transfer, e.g. checksum.md5 in the transfer metadata
+folder.
+"""
+
+# This file is part of Archivematica.
+#
+# Copyright 2010-2017 Artefactual Systems Inc. <http://artefactual.com>
+#
+# Archivematica is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Archivematica is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Archivematica.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import print_function, unicode_literals
+
+import datetime
+import os
+import subprocess
+import sys
+
+import django
+from django.db import transaction
+
+django.setup()
+from main.models import File
+
+from custom_handlers import get_script_logger
+from databaseFunctions import insertIntoEvents
+
+
+def concurrent_instances():
+    """Function docstring. N.B. Delete me."""
+    return 1
+
+
+logger = get_script_logger("archivematica.mcp.client.compare_hashes")
+
+
+class NoHashCommandAvailable(Exception):
+    """Provide feedback to the user if the checksum command cannot be found
+    for the provided checksum file.
+    """
+
+
+class PREMISFailure(Exception):
```

Comment:
I noticed this is raised in two different places so maybe it needs some context message.